projects/storefrontlib/src/lib/checkout/guards/delivery-mode-set.guard.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, UrlTree, Router } from '@angular/router';
+
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { CheckoutService, DeliveryMode } from '@spartacus/core';
+import { CheckoutConfig } from '../config/checkout-config';
+import { CheckoutStep } from '../config/model/checkout-step.model';
+import { CheckoutStepType } from '../config/default-checkout-config';
+
+@Injectable()
+export class DeliveryModeSetGuard implements CanActivate {
+  constructor(
+    private checkoutService: CheckoutService,
```

Comment:
Use CheckoutDetailsService instead of CheckoutService, because it will take care of things like loading checkout details before checking.